src/main/java/kr/co/vcnc/haeinsa/HaeinsaKeyValue.java

```diff
@@ -53,6 +53,18 @@ public int compare(HaeinsaKeyValue o1, HaeinsaKeyValue o2) {
         }
     };
 
+    public static final Comparator<HaeinsaKeyValue> REVERSE_COMPARATOR = new Comparator<HaeinsaKeyValue>() {
+        @Override
+        public int compare(HaeinsaKeyValue o1, HaeinsaKeyValue o2) {
+            return ComparisonChain.start()
+                .compare(o2.getRow(), o1.getRow(), new NullableComparator<>(Bytes.BYTES_COMPARATOR))
+                .compare(o1.getFamily(), o2.getFamily(), new NullableComparator<>(Bytes.BYTES_COMPARATOR))
+                .compare(o1.getQualifier(), o2.getQualifier(), new NullableComparator<>(Bytes.BYTES_COMPARATOR))
+                .compare(o2.getType().getCode() & 0xFF, o1.getType().getCode() & 0xFF)
```

Comment:
I think we are only reversing rows so we should also reverse getType() here?

Reply:
If I change this to  `o1.getType, o2.getType` my new test fails. We need to only swap the row order here http://github.com/Factual/haeinsa/blob/0e408f8fd42b65c38c4a2561fdd39c1014d7374f/src/main/java/kr/co/vcnc/haeinsa/HaeinsaKeyValue.java#L47-L52 but leave the rest the same including the type comparison order.

What happens is the scanner expects to see Puts before Deletes all else the same (row, family, qualifier) so we get this
```
11:25:28,742  WARN HaeinsaTable$ClientScanner - X kv b-1:data:phoneNumber:4 Put
11:25:28,742  WARN HaeinsaTable$ClientScanner - 1 kv b-1:data:phoneNumber:c DeleteColumn
```
And the delete is lost b/c of the type sort logic. (here it's sorted like you suggested.) *note: `4` and `c` are the hex value of put and deleteColumn. `X` is Max Long value of `getSequenceID`


Sorted and working in test, it looks like:
```
11:31:39,064  WARN HaeinsaTable$ClientScanner - 1 kv b-1:data:phoneNumber:c DeleteColumn
11:31:39,064  WARN HaeinsaTable$ClientScanner - X kv b-1:data:phoneNumber:4 Put
```

Reply:
Thanks for the explanation. LGTM